src/lib.rs

```diff
@@ -762,6 +762,61 @@ pub trait Filesystem {
     }
 }
 
+#[derive(Debug, Hash, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub struct Inode(u64);
+
+impl From<u64> for Inode {
+    fn from(ino: u64) -> Self {
+        Self(ino)
+    }
+}
+
+pub trait Filesystem2 {
+    fn init(&mut self, _req: &Request<'_>, _config: &mut KernelConfig) -> Result<(), c_int> {
+        Ok(())
+    }
+    fn destroy(&mut self, _req: &Request<'_>) {}
+    fn forget(&mut self, _req: &Request<'_>, _ino: Inode, _nlookup: u64) {}
+    fn getattr(
+        &mut self,
+        _req: &Request<'_>,
+        _ino: Inode,
+    ) -> Result<(&std::time::Duration, &FileAttr), c_int> {
```

Comment:
These references seem like they will make the implementation complicated. Have you tried this interface out to see out well it works?

Reply:
Yep, you are right! I tried before a similar API, but not with the reference.

In this case makes much more sense to get the value.

Reply:
Ya, I think in most cases it's best to return a value. Although, I think `read` and maybe `readlink` would need to return references as a potential optimization to avoid copying memory. Can you think of a better way to deal with that?

I rely on that in fleetfs, so that the TCP client can reuse a single thread local buffer: https://github.com/fleetfs/fleetfs/blob/84fe3476681c268afcb69d039b1cb3cf4dbd56c7/src/fuse_adapter.rs#L661